processor/batchprocessor/batch_processor.go

```diff
@@ -307,9 +307,9 @@ func (mb *multiShardBatcher) consume(ctx context.Context, data any) error {
 	}
 	aset := attribute.NewSet(attrs...)
 
+	mb.lock.Lock()
```

Comment:
This is a very poor solution to the problem. A way simpler and correct solution would be to just re-check the map and leave fast if already there after capturing the lock.